info_encoding.md

```diff
@@ -0,0 +1,143 @@
+# Encoding of the `info` Field in Derivations
+
+The key derivation should include all information of the context
+of the derivation.
+This includes information that is valid only for this particular message,
+as well as the intended use of the key.
+
+Once we have agreed on a suitable selection of values to include, we
+also must encode them in such a way that it can not collide with
+the honest encoding of a different set of values.
+Obviously, an adversary can perform a different encoding, but this
+does not matter, as we assume they can not trick the user into performing
+off-standard derivations or use an off-standard encoding for their derivations.
+Implementations must make sure this assumption holds.
+
+## Notation
+
+In this document, we write lists as comma seperated value, enclosed
+in parentheses. Therefore a list of values 1, 2 and 3 would be written
+as `(1, 2, 3)`.
+Binary concatenation is `||`.
+
+## Requirements
+
+We require the format to support lists of buffers.
+This is enough for most simple operations, and can be extended to
+key-value stores.
+
+To get a simpler encoding, we do not require schemalessness.
```

Comment:
is this a double negative. Could we just say 

> To get a simpler encoding, we define a schema for encoding the info field

Reply:
Hm, to me schemalessness is a special word... How about "we require the user to adhere to a schema"?

Reply:
Hm, that makes the sentence afterwards weird.

The thing is that we don't want to define the schema ourselves, we want to user to know what they are doing.

Reply:
How about:
> A property of many structured data encodings is schemalessness. To get a simpler encoding, we do not require this property. Therefore, derivation steps need to define a schema for the encoding of the info field.

Reply:
It's just an odd structure to me. Can we go : 

> A property of many structured data encodings is schemalessness. 
> To get a simpler encoding, we are instead requiring a schema. 
> The definition of a schema for each encoding (i.e. what's in the `list` of info, and what the order is) is therefore required. 
> In this module, see `DeriveSecret` which is using this SLP